processing/src/main/java/org/apache/druid/segment/data/GenericIndexed.java

```diff
@@ -82,7 +82,7 @@
  *
  * @see GenericIndexedWriter
  */
-public class GenericIndexed<T> implements CloseableIndexed<T>, Serializer
+public abstract class GenericIndexed<T> implements CloseableIndexed<T>, Serializer
```

Comment:
side thought, i wonder if we should remove Serializer from GenericIndexed, it seems to only be used in IndexMergerV9 to write out column lists, which seem like just as easily could be using GenericIndexedWriter like everything else.

Reply:
Probably, I think that makes sense. I agree with the spirit of these two comments, which is that GenericIndexed is kind of over-complicated.